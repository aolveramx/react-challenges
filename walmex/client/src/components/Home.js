import { Fragment, useState } from 'react'
import useFetch from '../hooks/useFetch'

import { Table } from './styles/Home.styled'
import { StyledSelect } from './styles/Dropdown.styled'

const Home = () => {
  const shopingCart = [
    'carrot',
    'apple',
    'grapes',
    'cake',
    'tv',
    'cracker',
    'chips',
    'ham',
  ]

  const [productURL, setProductURL] = useState('')

  const select = shopingCart.map(add => add)

  const handleClick = e => {
    const url = `/users/age?product=${e.target.value}`
    setProductURL(url)
  }

  const { isLoading, data, error } = useFetch(productURL)
  console.log(data)

  return (
    <Fragment>
      {isLoading && <div>Loading...</div>}
      {error && <div>{console.log(error)}</div>}
      {data && (
        <Table>
          <thead>
            <tr>
              <th>Age</th>
              <th>Count</th>
            </tr>
          </thead>
          <tbody>
            {data.map((d, index) => {
              let key = Object.keys(d).pop()
              return (
                <tr key={index}>
                  <td>{key}</td>
                  <td>{d[key]}</td>
                </tr>
              )
            })}
          </tbody>
        </Table>
      )}
      <StyledSelect onChange={e => handleClick(e)}>
        {select.map((item, index) => (
          <option value={item} key={index}>
            {item}
          </option>
        ))}
      </StyledSelect>
    </Fragment>
  )
}

export default Home
