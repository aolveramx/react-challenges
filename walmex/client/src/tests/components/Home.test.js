import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import Home from "../../components/Home"

describe('Tests in <Home />', () => {

 test('should render <Home />', () => {
   const wrapper = shallow(<Home />)
   expect(wrapper).toMatchSnapshot()
 });

})