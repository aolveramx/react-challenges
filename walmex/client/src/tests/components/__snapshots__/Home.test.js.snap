// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Home /> should render <Home /> 1`] = `
<Fragment>
  <div>
    Loading...
  </div>
  <styled.select
    onChange={[Function]}
  >
    <option
      key="0"
      value="carrot"
    >
      carrot
    </option>
    <option
      key="1"
      value="apple"
    >
      apple
    </option>
    <option
      key="2"
      value="grapes"
    >
      grapes
    </option>
    <option
      key="3"
      value="cake"
    >
      cake
    </option>
    <option
      key="4"
      value="tv"
    >
      tv
    </option>
    <option
      key="5"
      value="cracker"
    >
      cracker
    </option>
    <option
      key="6"
      value="chips"
    >
      chips
    </option>
    <option
      key="7"
      value="ham"
    >
      ham
    </option>
  </styled.select>
</Fragment>
`;
